virtcontainers/utils/utils_linux.go

```diff
@@ -76,14 +76,16 @@ func FindContextID() (*os.File, uint64, error) {
 	// Looking for the first available context ID.
 	for cid := contextID; cid <= maxUInt; cid++ {
 		if err := ioctlFunc(vsockFd.Fd(), ioctlVhostVsockSetGuestCid, cid); err == nil {
-			return vsockFd, cid, nil
+			vsockFd.Close()
```

Comment:
Is it better combine those three fd close into a defer function?

Reply:
Yes. I still need to see in the QEMU code if it uses the fds. I may need to close the fd's at the point where I hand it off to firecracker to prevent a race condition. However closing it will trigger a race.